### PR TITLE
[test][doc] Spark time travel with data frame test and doc

### PR DESCRIPTION
**Pull request by hililiwei**
### Purpose

Linked issue: close #xxx

add test and doc for spark time travel with data frame 

### Tests

org.apache.paimon.spark.SparkTimeTravelWithDataFrameITCase

### Documentation

Add quering with data frame in spark3.md

### CODE REVIEW

File: docs/content/engines/spark3.md [hunk 1]
```diff
@@ -205,6 +205,50 @@ dataset.createOrReplaceTempView("my_table")
...
+-- read the snapshot from specified timestamp in unix seconds
+SELECT * FROM t TIMESTAMP AS OF 1678883047;
+
+-- read tag 'my-tag'
+SELECT * FROM t VERSION AS OF 'my-tag';
+```
+
+### DataFrame
+
+To select a specific table snapshot or the snapshot at some time in the DataFrame API, Paimon supports:
```
Comment (JingsongLi) [inline docs/content/engines/spark3.md:231]: Can we just put this into query-table page?

**Review by JingsongLi — COMMENTED** [review]
Thanks @hililiwei , left one comment

**Review by JingsongLi — COMMENTED** [review]
Thanks @hililiwei , left one comment

File: docs/content/engines/spark3.md (hunk 1 above)
Comment (hililiwei) [reply docs/content/engines/spark3.md]: Done.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**